drivers/soundwire/intel.c

```diff
@@ -883,16 +939,31 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd,
 	}
 
 	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_RESUME:
+		dev_err(dai->dev, "%s: %s: resume\n", __func__, dai->name);
+		/* fallthrough */
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
-	case SNDRV_PCM_TRIGGER_RESUME:
 		ret = sdw_enable_stream(dma->stream);
 		break;
 
-	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		dev_err(dai->dev, "%s: %s: suspend\n", __func__, dai->name);
+		dma->suspend_reinitialize = true;
+		/* fallthrough */
+	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 	case SNDRV_PCM_TRIGGER_STOP:
 		ret = sdw_disable_stream(dma->stream);
+
+		if (!ret) {
+			err = sdw_deprepare_stream(dma->stream);
```

Comment:
I still think it is better to call sdw_deprepare_stream() in .hw_free().  I saw intel_prepare() is trigger twice without intel_trigger() when pulseaudio is enabled.